dev-util/patchelf: Add define for DT_MIPS_XHASH

LinuxUserGD opened the pull request:
dev-util/patchelf: Add define for DT_MIPS_XHASH

For musl libc the bundled `elf.h` header file is replaced with `DT_MIPS_XHASH` define using regex
(alternative approach to https://bugs.gentoo.org/860888#c14 which patches the if branch directly).

Closes: https://bugs.gentoo.org/860888

chewi commented:
I'd much rather do this with a patch and `eapply`. That sed is quite awkward and could easily break.

LinuxUserGD commented:
> I'd much rather do this with a patch and `eapply`. That sed is quite awkward and could easily break.

Ok, I've added the patch after `rm` command.
Also removed `use elibc_musl` so it could apply to different libc `elf.h` headers too (or should I keep the condition?).

chewi commented:
Ah sorry, if you're not going to make it conditional then it should be applied with this instead:

```bash
PATCHES=(
	"${FILESDIR}"/${PN}-glibc-dt-mips-xhash.patch
)

src_prepare() {
	rm src/elf.h || die
	default
}
```

`default` is where it applies the patches, so it'll do that after the `rm`. I think making it unconditional is fine and we generally prefer that anyway.

I must admit I'm not keen on `#include <../include/elf.h>`. It feels like a huge hack. I would prefer `#include_next <elf.h>`. It's supported by at least GCC and Clang. Clang emits a warning about it, but I'm not sure why, and I think I still prefer it over the alternative.
